crates/sargon/src/system/sargon_os/sargon_os_transactions.rs

```diff
@@ -0,0 +1,382 @@
+use crate::prelude::*;
+use std::time::Duration;
+
+// ==================
+// Submit Transaction
+// ==================
+#[uniffi::export]
+impl SargonOS {
+    /// Submits a notarized transaction payload to the network.
+    pub async fn submit_transaction(
+        &self,
+        notarized_transaction: NotarizedTransaction,
+    ) -> Result<IntentHash> {
+        let network_id = self.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+        gateway_client
+            .submit_notarized_transaction(notarized_transaction)
+            .await
+    }
+
+    /// Submits a compiled transaction payload to the network.
+    pub async fn submit_compiled_transaction(
+        &self,
+        compiled_notarized_intent: CompiledNotarizedIntent,
+    ) -> Result<IntentHash> {
+        self.submit_transaction(compiled_notarized_intent.decompile())
+            .await
+    }
+}
+
+// ==================
+// Poll Transaction Status
+// ==================
+#[uniffi::export]
+impl SargonOS {
+    /// Polls the state of a Transaction until we can determine its `TransactionStatus`.
+    pub async fn poll_transaction_status(
+        &self,
+        intent_hash: IntentHash,
+    ) -> Result<TransactionStatus> {
+        let network_id = self.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+
+        // The delay increment is set to 1 second in production, but 0 seconds in tests.
+        // This will make the tests run with 0s delay, while the production code will have a 2s delay on first call,
+        // a 3s delay on second call, 4s on third and so on.
+        #[cfg(test)]
+        const DELAY_INCREMENT: u64 = 0;
```

Comment:
We might wanna unit test  the durations slept (a vec, and the length of which will be the amount we polled (plus or minus 1 :P )), to assert we correctly increase duration between each loop.

Thus you must use a non zero value for `cfg(test)` so I propse we switch to milliseconds instead of seconds. and you can use 1ms for tests...

 `poll_transaction_status` into two methods:
```rust
/// Live impl, but retaining some statistics about how many times we did poll, which
/// are useful for tests
fn poll_transaction_status_with_stats(..) -> Result<(TransasctionStatus, Vec<u64>)> // `Vec<u64>` is duration slept between each poll in milliseconds
```

and  a method `poll_transaction_status` which signature is identical to what you have here, which calls `poll_transaction_status_with_stats` but ignores the stats (return type `Result<TransasctionStatus>`.